### ISSUE TITLE
yield in loop confuses branch coverage

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#324]
```
def gen(inp):
    for n in inp:
        yield (i * 2 for i in range(n))

list(gen([1,2,3]))
```

Running this with --branch gives strange results.  "2 partial branches", but only line 3 is marked, and it's marked with "2 exit".

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/324

**Comment by nedbat** [comment]
*Original comment by* **Mickie Betz (Bitbucket: [mbetz08](https://bitbucket.org/mbetz08), GitHub: [mbetz08](https://github.com/mbetz08))**

----------------------------------------

I was able to reproduce the issue with the input above.  Here's the report output:

Name           Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------
issue-324.py       4      0      4      2    75%   5->-5, 5->4

The html output looks like:

![Screen Shot 2015-04-13 at 12.43.01 PM.png](https://bitbucket.org/repo/Ee7X/images/231710083-Screen%20Shot%202015-04-13%20at%2012.43.01%20PM.png)

**Comment by nedbat** [comment]
Fixed in [2e992bf44abaf0d3c78134a4eb9724f015d23e00 (bb)](https://bitbucket.org/ned/coveragepy/commits/2e992bf44abaf0d3c78134a4eb9724f015d23e00)
